### PR TITLE
 SAR-4611:Create new controller for property cash/accruals and amend …

### PR DESCRIPTION
**Pull request by scullythornton1**
…the old flow

### CODE REVIEW

File: it/incometax/business/PropertyAccountingMethodControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+          httpStatus(SEE_OTHER),
+          redirectURI(checkYourAnswersURI)
+        )
+      }
+    }
+
+    "not select an option on the property accounting method page" in {
+      Given("I setup the Wiremock stubs")
+      AuthStub.stubAuthSuccess()
+//      KeystoreStub.stubKeystoreData(keystoreData(propertyAccountingMethod = Some(testPropertyAccountingMethodCash)))
```
Comment (MarjanMubarok) [inline it/incometax/business/PropertyAccountingMethodControllerISpec.scala]: remove?

File: app/incometax/business/controllers/BusinessAccountingMethodController.scala [hunk 2]
```diff
@@ -67,6 +67,9 @@ class BusinessAccountingMethodController @Inject()(val baseConfig: BaseControlle
         formWithErrors => view(accountingMethodForm = formWithErrors, isEditMode = isEditMode).map(view => BadRequest(view)),
         accountingMethod => {
           keystoreService.saveAccountingMethod(accountingMethod) map (_ => if (isEditMode || appConfig.eligibilityPagesEnabled) {
+            if (appConfig.propertyCashOrAccrualsEnabled)
+              Redirect(incometax.business.controllers.routes.PropertyAccountingMethodController.show())
+            else
```
Comment (MarjanMubarok) [inline app/incometax/business/controllers/BusinessAccountingMethodController.scala]: Indentation with the nested ifs.

File: it/helpers/ComponentSpecBase.scala [hunk 3]
```diff
@@ -330,6 +332,16 @@ trait ComponentSpecBase extends UnitSpec
       )
     }
 
+    def submitPropertyAccountingMethod(inEditMode: Boolean, request: Option[AccountingMethodPropertyModel]): WSResponse = {
+      val uri = s"/business/accounting-method-property?editMode-=$inEditMode"
+      post(uri)(
+        request.fold(Map.empty[String, Seq[String]])(
+          model =>
+            AccountingMethodPropertyForm.accountingMethodPropertyForm.fill(model).data.map { case (k, v) => (k, Seq(v)) }
```
Comment (david-walters-HMRC) [inline it/helpers/ComponentSpecBase.scala]: what is K and V?, more descriptive variable names

File: test/incometax/business/forms/AccountingPeriodDateFormSpec.scala [hunk 4]
```diff
@@ -116,7 +116,7 @@ class AccountingPeriodDateFormSpec extends PlaySpec with OneAppPerTest {
           invalid fieldErrorIs MessageLookup.Error.EndDate.invalid
           invalid summaryErrorIs MessageLookup.Error.EndDate.invalid
 
-          val invalidTest = accountingPeriodDateForm.bind(DataMap.date(endDate)("29", "2", LocalDate.now.getDayOfYear.toString))
+          val invalidTest = accountingPeriodDateForm.bind(DataMap.date(endDate)("34", "2", LocalDate.now.getYear.toString))
```
Comment (david-walters-HMRC) [inline test/incometax/business/forms/AccountingPeriodDateFormSpec.scala]: why does this need to be dynamic for the date?, just hard-code the year.

File: it/helpers/ComponentSpecBase.scala (hunk 3 above)
Comment (scullythornton1) [reply it/helpers/ComponentSpecBase.scala]: in order to keep it consistent so I will leave it as k and v.

File: test/incometax/business/forms/AccountingPeriodDateFormSpec.scala (hunk 4 above)
Comment (scullythornton1) [reply test/incometax/business/forms/AccountingPeriodDateFormSpec.scala]: has changed them to hard-code

File: it/incometax/business/PropertyAccountingMethodControllerISpec.scala (hunk 1 above)
Comment (scullythornton1) [reply it/incometax/business/PropertyAccountingMethodControllerISpec.scala]: removed

**Pull request merged**